tests/github/experimental/workspace/testcase_private_attributes_initialization_with_both_options/src/testflow_privateattributes.py

```diff
@@ -171,17 +197,44 @@ def validate_collab_private_attr(self, private_attr, step_name):
             )
 
 
-def validate_agg_private_attrs(self, private_attr_1, private_attr_2, step_name):
+def validate_collab_private_attrs(self, step_name, **private_attrs_kwargs):
+
+    """
+        Validate that collaborators can only access their own attributes
+
+        Args:
+            step_name: Name of the step being validated 
+            private_attr_kwargs: Keyword arguments specifying the names of private attributes for the aggregator and collaborators.
+    """
+    agg_attrs = private_attrs_kwargs.get('aggr',[])
+    collab_attrs = private_attrs_kwargs.get('collabs', [])
+
     # Collaborator should only be able to access its own attributes
-    if not hasattr(self, private_attr_1) or not hasattr(self, private_attr_2):
+ 
+    # check for missing collaborators attributes 
+    inaccessible_collab_attrs = [attr for attr in collab_attrs if not hasattr(self,attr)]
+
+    if inaccessible_collab_attrs:
         TestFlowPrivateAttributes.ERROR_LIST.append(
-            step_name + "collab_attributes_not_found"
+            step_name + " collab_attributes_not_found"
         )
         print(
             f"{bcolors.FAIL} ... Attribute test failed in {step_name} - Collab "
             + f"private attributes not accessible {bcolors.ENDC}"
         )
+    # check for aggregator private attributes that should not be accessible 
+    breached_agg_attr = [attr for attr in agg_attrs if hasattr(self, attr) is True]
```

Comment:
nit - you could shorten the condition to simply `agg_attrs if hasattr(self, attr)`

Reply:
Thanks for the suggestion, we have simplified the filtering condition as mentioned